Problem with fr"...\ "

See here: https://travis-ci.com/theislab/scanpydoc/jobs/169231095

```py
tests/test_elegant_typehints.py:None (tests/test_elegant_typehints.py)
3.5/lib/python3.5/site-packages/_pytest/python.py:445: in _importtestmodule
    mod = self.fspath.pyimport(ensuresyspath=importmode)
3.5/lib/python3.5/site-packages/py/_path/local.py:668: in pyimport
    __import__(modname)
<frozen importlib._bootstrap>:968: in _find_and_load
    ???
<frozen importlib._bootstrap>:957: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:664: in _load_unlocked
    ???
<frozen importlib._bootstrap>:634: in _load_backward_compatible
    ???
3.5/lib/python3.5/site-packages/_pytest/assertion/rewrite.py:308: in load_module
    six.exec_(co, mod.__dict__)
tests/test_elegant_typehints.py:3: in <module>
    from scanpydoc.elegant_typehints import format_annotation, _format_terse, _format_full
E     File "/home/phil/Dev/Python/scanpydoc/scanpydoc/elegant_typehints.py", line 115
E       fr":annotation-terse:`{_escape(_format_terse(annotation))}`\ "
E                                                                    ^
E   SyntaxError: invalid syntax
```

--- a/future_fstrings.py
+++ b/future_fstrings.py
@@ -153,7 +153,7 @@ def _fstring_parse_outer(s, i, level, parts, exprs):
 
 
 def _is_f(tokens, i):
-    return i >= 0 and tokens[i].name == 'NAME' and tokens[i].src.lower() == 'f'
+    return i >= 0 and tokens[i].name == 'NAME' and 'f' in tokens[i].src.lower()
 
 
 def _make_fstring(tokens):
@@ -172,6 +172,8 @@ def _make_fstring(tokens):
                 _fstring_parse_outer(token.src, 0, 0, parts, exprs)
             except SyntaxError as e:
                 raise TokenSyntaxError(e, tokens[i - 1])
+            if 'r' in tokens[i - 1].src.lower():
+                parts = [s.replace('\\', '\\\\') for s in parts]
             token = token._replace(src=''.join(parts))
         elif token.name == 'STRING' and not _is_f(tokens, i - 1):
             new_src = token.src.replace('{', '{{').replace('}', '}}')

--- a/tests/future_fstrings_test.py
+++ b/tests/future_fstrings_test.py
@@ -71,6 +71,11 @@ def test_strings_quoting_variables():
     assert f"""hello {'''hi '" hello'''}""" == 'hello hi \'" hello'
 
 
+def test_raw_strings():
+    assert fr'hi\ {1}' == r'hi\ 1'
+    assert fr'\n {1}' == r'\n 1'
+
+
 def test_sequence_literals():
     assert f'hello {[1, 2, 3][0]}' == 'hello 1'
     assert f'hello {sorted({1, 2, 3})[0]}' == 'hello 1'

